src/core/tensor/math_kernel.cu

```diff
@@ -184,6 +184,20 @@ __global__ void KernelAbs(const size_t n, const float *in, float *out) {
   }
 }
 
+__global__ void KernelAsType2(const size_t n, const float *in, int *out) {
+  for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < n;
+       i += blockDim.x * gridDim.x) {
+    out[i] = in[i];
```

Comment:
do you need the cast operation like `out[i]=int(in[i])?

Reply:
I think it is working as implicit casting now

Reply:
better make it explicit.

Reply:
thank you for the suggestion. it is explicited now.